apps/dashboard/src/pages/Images.tsx

```diff
@@ -37,33 +37,71 @@ const Images: React.FC = () => {
 
   const { selectedOrganization, authenticatedUserHasPermission } = useSelectedOrganization()
 
+  const [pagination, setPagination] = useState({
+    pageIndex: 0,
+    pageSize: 10,
+    totalItems: 0,
+    totalPages: 0,
+  })
+
   const fetchImages = useCallback(
-    async (showTableLoadingState = true) => {
+    async (showTableLoadingState = true, pageIndex = pagination.pageIndex, pageSize = pagination.pageSize) => {
       if (!selectedOrganization) {
         return
       }
       if (showTableLoadingState) {
         setLoadingTable(true)
       }
       try {
-        const response = (await imageApi.getAllImages(selectedOrganization.id)).data
+        const response = (await imageApi.getAllImages(selectedOrganization.id, pageSize, pageIndex + 1)).data
+
         setImages(response.items)
+        setPagination((prev) => ({
+          ...prev,
+          totalItems: response.total,
+          totalPages: response.totalPages,
+        }))
       } catch (error) {
         handleApiError(error, 'Failed to fetch images')
       } finally {
         setLoadingTable(false)
       }
     },
-    [imageApi, selectedOrganization],
+    [imageApi, selectedOrganization, pagination.pageIndex, pagination.pageSize],
+  )
+
+  const handlePaginationChange = useCallback(
+    ({ pageIndex, pageSize }: { pageIndex: number; pageSize: number }) => {
+      setPagination((prev) => ({
+        ...prev,
+        pageIndex,
+        pageSize,
+      }))
+      fetchImages(true, pageIndex, pageSize)
+    },
+    [fetchImages],
   )
 
   useEffect(() => {
     fetchImages()
-  }, [fetchImages])
+  }, [fetchImages, pagination.pageIndex, pagination.pageSize])
 
   useEffect(() => {
     const handleImageCreatedEvent = (image: ImageDto) => {
-      setImages((prev) => [image, ...prev])
+      if (pagination.pageIndex === 0) {
+        setImages((prev) => {
+          const newImages = [image, ...prev]
```

Comment:
@unsuman We should add the following `if` block before constructing the new array just to avoid inconsistencies:

```suggestion
          if (prev.some((i) => i.id === image.id)) {
            return prev
          }
          const newImages = [image, ...prev]
```